### PR TITLE
Add Excel storage on S3

### LINKED ISSUE #778: Implement Excel file storage
Breakout of #772 

### Background

We need to configure our Django app to use our private S3 bucket as its default storage mechanism.

The work for this ticket should focus solely on setting up the file upload and storage pipeline. File validation and scanning will be covered in a separate ticket.

### Acceptance Criteria

- [x] The app is configured to use our private S3 bucket as its default file storage mechanism
- [x] The local development configuration is updated to use `localstack` to emulate S3
- [ ] The local development documentation is updated accordingly
- [ ] When an ExcelFile is added via Django admin, the file is stored in the private S3 bucket and not the local filesystem

### PR DESCRIPTION
**Pull request by geekygirlsarah**
Resolves #778 

This adds in the file upload and storage piepline for uploading Excel documents into S3 (but doesn't add validation/scanning on it). 

* Adds S3 private and public buckets to Terraform
* Adds settings to fetch Cloud.gov's settings for the buckets
* Adds LocalStack in Docker-Compose to support S3 uploads for local testing
* Sets the default media URLs in Django to be the S3 private bucket

### CODE REVIEW

**Comment by geekygirlsarah** [comment]
There's now a dependency whack-a-mole issue of djlint needing PyYAML >=6.0,<7.0, and aws-cli needing PyYAML <5.5. I'm not sure how to resolve this at the moment but brought it up with the team.

File: backend/report_submission/storages.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.conf import settings
+from storages.backends.s3boto3 import S3Boto3Storage
+
+
+class S3PrivateStorage(S3Boto3Storage):
```
Comment (timoballard) [inline backend/report_submission/storages.py:5]: Can we add a comment here explaining that this class override exists to get around the fact that `django-storages` by default references the `AWS_STORAGE_BUCKET_NAME` environment variable. Without this (or something similar), we'd be unable to have multiple S3 buckets (e.g. one public, one private)

Comment (geekygirlsarah) [reply backend/report_submission/storages.py:5]: Added!

File: backend/run.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+if [[ -n "${ENV}" ]]; then
+  echo "Environment set as: ${ENV}"
+else
+  echo "No environment variable `ENV` is set!"
+fi;
+
+export AWS_ACCESS_KEY_ID=test
+export AWS_SECRET_ACCESS_KEY=test
+
+#if [[ "${ENV}" == "LOCAL" ]] || [[ "${ENV}" == "TESTING" ]]; then
```
Comment (timoballard) [inline backend/run.sh:12]: hangover debugging code - need to put this conditional back in 😁

Comment (geekygirlsarah) [reply backend/run.sh]: Oops... fixed!

**Review by timoballard — APPROVED** [review]
lgtm!

**Pull request merged**